src/include/txn_inline.h

```diff
@@ -1295,24 +1261,43 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
 static inline int
 __wt_txn_search_check(WT_SESSION_IMPL *session)
 {
-    WT_BTREE *btree;
     WT_TXN *txn;
+    uint32_t flags;
+    const char *name;
 
-    btree = S2BT(session);
     txn = session->txn;
+    flags = session->dhandle->ts_flags;
+    name = session->dhandle->name;
 
-    /*
-     * If the user says a table should always use a read timestamp, verify this transaction has one.
-     * Same if it should never have a read timestamp.
-     */
-    if (!F_ISSET(S2C(session), WT_CONN_RECOVERING) &&
-      FLD_ISSET(btree->dhandle->ts_flags, WT_DHANDLE_ASSERT_TS_READ_ALWAYS) &&
-      !F_ISSET(txn, WT_TXN_SHARED_TS_READ))
-        WT_RET_MSG(session, EINVAL, "read_timestamp required and none set on this transaction");
-    if (FLD_ISSET(btree->dhandle->ts_flags, WT_DHANDLE_ASSERT_TS_READ_NEVER) &&
-      F_ISSET(txn, WT_TXN_SHARED_TS_READ))
-        WT_RET_MSG(
-          session, EINVAL, "no read_timestamp required and timestamp set on this transaction");
+    /* Skip for any file not configured for checking. */
+    if (!LF_ISSET(WT_DHANDLE_ASSERT_TS_READ_ALWAYS | WT_DHANDLE_ASSERT_TS_READ_NEVER |
+          WT_DHANDLE_VERBOSE_TS_READ_ALWAYS | WT_DHANDLE_VERBOSE_TS_READ_NEVER))
+        return (0);
+
+    /* Skip during recovery. */
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING))
+        return (0);
+
+    /* Verify if the table should always or never use a read timestamp. */
+    if (LF_ISSET(WT_DHANDLE_ASSERT_TS_READ_ALWAYS | WT_DHANDLE_VERBOSE_TS_READ_ALWAYS) &&
```

Comment:
Later we are verifying the flags and raised the error message, can we remove the flag checks here?

Reply:
Yes.

This check isn't correct anyway, and there's no real need for it. I did a similar (but hopefully correct), check in `__txn_timestamp_usage_check()`, because there's a log of additional work in that function we'd have to do even if we're not configured for assert or verbose, and that's not true here.